src/scene/renderer/forward-renderer.js

```diff
@@ -32,69 +29,73 @@ import {
     PROJECTION_PERSPECTIVE
 } from '../constants.js';
 import { Material } from '../materials/material.js';
-import { Mesh } from '../mesh.js';
-import { MeshInstance } from '../mesh-instance.js';
 import { LayerComposition } from '../composition/layer-composition.js';
-import { ShadowRenderer } from './shadow-renderer.js';
 import { Camera } from '../camera.js';
 import { GraphNode } from '../graph-node.js';
 
-var shadowCamView = new Mat4();
-var shadowCamViewProj = new Mat4();
+import { ShadowRenderer } from './shadow-renderer.js';
+import { StaticMeshes } from './static-meshes.js';
+
+const shadowCamView = new Mat4();
+const shadowCamViewProj = new Mat4();
 
-var viewInvMat = new Mat4();
-var viewMat = new Mat4();
-var viewMat3 = new Mat3();
-var viewProjMat = new Mat4();
-var projMat;
+const viewInvMat = new Mat4();
+const viewMat = new Mat4();
+const viewMat3 = new Mat3();
+const viewProjMat = new Mat4();
+let projMat;
 
-var flipYMat = new Mat4().setScale(1, -1, 1);
-var flippedViewProjMat = new Mat4();
-var flippedSkyboxProjMat = new Mat4();
+const flipYMat = new Mat4().setScale(1, -1, 1);
+const flippedViewProjMat = new Mat4();
+const flippedSkyboxProjMat = new Mat4();
 
-var viewInvL = new Mat4();
-var viewInvR = new Mat4();
-var viewL = new Mat4();
-var viewR = new Mat4();
-var viewPosL = new Vec3();
-var viewPosR = new Vec3();
-var projL, projR;
-var viewMat3L = new Mat3();
-var viewMat3R = new Mat3();
-var viewProjMatL = new Mat4();
-var viewProjMatR = new Mat4();
+const viewInvL = new Mat4();
+const viewInvR = new Mat4();
+const viewL = new Mat4();
+const viewR = new Mat4();
+const viewPosL = new Vec3();
+const viewPosR = new Vec3();
+let projL, projR;
+const viewMat3L = new Mat3();
+const viewMat3R = new Mat3();
+const viewProjMatL = new Mat4();
+const viewProjMatR = new Mat4();
 
-var worldMatX = new Vec3();
-var worldMatY = new Vec3();
-var worldMatZ = new Vec3();
+const worldMatX = new Vec3();
+const worldMatY = new Vec3();
+const worldMatZ = new Vec3();
 
-var tempSphere = new BoundingSphere();
-var boneTextureSize = [0, 0, 0, 0];
-var boneTexture, instancingData, modelMatrix, normalMatrix;
+const tempSphere = new BoundingSphere();
+const boneTextureSize = [0, 0, 0, 0];
+let boneTexture, instancingData, modelMatrix, normalMatrix;
 
-var keyA, keyB;
+let keyA, keyB;
 
-var _autoInstanceBuffer = null;
+let _autoInstanceBuffer = null;
 
-var _skinUpdateIndex = 0;
+let _skinUpdateIndex = 0;
 
-var _tempMaterialSet = new Set();
+const _tempMaterialSet = new Set();
 
 /**
  * @class
  * @name ForwardRenderer
  * @classdesc The forward renderer render scene objects.
  * @description Creates a new forward renderer object.
+ * @hideconstructor
  * @param {GraphicsDevice} graphicsDevice - The graphics device used by the renderer.
+ * @param {Scene} A - scene for the rendering.
  */
 class ForwardRenderer {
-    constructor(graphicsDevice) {
```

Comment:
So this is interesting to me. Before, I guess the API suggested you could do:

```
const sceneA = new pc.Scene();
const sceneB = new pc.Scene();

renderer.render(sceneA);
renderer.render(sceneB);
```

But this explicitly creates a 1-2-1 relationship between the scene and the renderer. Would be great to get your thoughts on this.

Reply:
It's a valid point @willeastcott , and so I've reverted this one change. At some point in the future it would certainly be nice to support multiple scenes this way. At the moment this would not work, as the scene is referenced in few other places  like batch manager, lightmapper - these would be easy to fix, but also by Materials and Lights, which mark the layer composition dirty when they change - this could be harder to fix.

Reply:
Yep, I did suspect that the rendering of multiple scenes had broken over time. But yeah, I think it's a good call to roll this change back. Nice.